src/igvfd/types/file_set.py

```diff
@@ -125,12 +125,53 @@ class AnalysisSet(FileSet):
     item_type = 'analysis_set'
     schema = load_schema('igvfd:schemas/analysis_set.json')
     embedded_with_frame = FileSet.embedded_with_frame + [
-        Path('input_file_sets', include=['@id', 'accession', 'aliases'])
+        Path('input_file_sets', include=['@id', 'accession', 'aliases', 'file_set_type'])
     ]
     audit_inherit = FileSet.audit_inherit
     set_status_up = FileSet.set_status_up + []
     set_status_down = FileSet.set_status_down + []
 
+    @calculated_property(
+        schema={
+            'title': 'Summary',
+            'type': 'string',
+            'notSubmittable': True,
+        }
+    )
+    def summary(self, request, file_set_type, input_file_sets=[]):
+        sentence = f'{file_set_type}'
+        inspected_filesets = set()
+        assay_terms = set()
+        fileset_types = set()
+        if input_file_sets:
+            filesets_to_inspect = set(input_file_sets.copy())
+
+            while filesets_to_inspect:
+                input_fileset = filesets_to_inspect.pop()
+                if input_fileset not in inspected_filesets:
+                    inspected_filesets.add(input_fileset)
+                    fileset_object = request.embed(input_fileset, '@@object?skip_calculated=true')
+                    if input_fileset.startswith('/measurement-sets/'):
+                        if 'preferred_assay_title' in fileset_object:
+                            assay_terms.add(fileset_object['preferred_assay_title'])
+                        else:
+                            assay_terms.add(request.embed(fileset_object['assay_term'],
+                                            '@@object?skip_calculated=true')['term_name'])
+                    elif not input_fileset.startswith('/analysis-sets/'):
+                        fileset_types.add(fileset_object['file_set_type'])
+                    elif (input_fileset.startswith('/analysis-sets/') and
+                          fileset_object.get('input_file_sets', False)):
+                        filesets_to_inspect.update(set(fileset_object.get('input_file_sets')))
```

Comment:
for `input_file_set` in `input_file_sets`, make sure it is not in `inspected_filesets` first, than you can add it to `filesets_to_inspect`

Reply:
fixed